crates/admin/src/cluster_controller/service.rs

```diff
@@ -307,9 +307,16 @@ impl<T: TransportConnect> Service<T> {
                 },
                 Ok(cluster_state) = cluster_state_watcher.next_cluster_state() => {
                     self.observed_cluster_state.update(&cluster_state);
-                    state.update(&self).await?;
+                    // todo quarantine this cluster controller if errors re-occur too often so that
+                    //  another cluster controller can take over
+                    if let Err(err) = state.update(&self).await {
+                        warn!("Failed to update cluster state. This can impair the overall cluster operations: {:?}", err);
+                        continue;
+                    }
 
-                    state.on_observed_cluster_state(&self.observed_cluster_state).await?;
+                    if let Err(err) = state.on_observed_cluster_state(&self.observed_cluster_state).await {
+                        warn!("Failed to handle observed cluster state. This can impair the overall cluster operations: {:?}", err);
```

Comment:
```suggestion
                        warn!("Failed to handle observed cluster state. This can impair the overall cluster operations: {}", err);
```